src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,32 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction() {
+                return null;
```

Comment:
I prefer the symmetry actually and returning null for one of the methods does not look nice. 

Maybe it makes sense to check not for the null but for the `USE_FROM_SERVICE`?

If you do not want to parametrize inside the checking code, one could crate a constant set to `USE_FROM_SERVICE` and check against that constant.

What do you think?